Created an EventController that accepts the EventTypeId

Jamie-Unosquare opened the pull request:
Use this with the type ID rather than using specific controllers.
The old controllers still work so the front end can use these while moving over to using the new /Event/ endpoint, they will need pass back the EventTypeId with all their calls from now on to use this.

Jamie-Unosquare committed:
Created an EventController that accepts the EventTypeId

Use this with the type ID rather than using specific controllers.
The old controllers still work so the front end can use these while moving over to using the new /Event/ endpoint, they will need pass back the EventTypeId with all their calls from now on to use this.

nmcmahon-unosquare commented on AdminCore.WebApp/Controllers/EventController.cs:
```diff
@@ -0,0 +1,276 @@
+        return Ok(_mapper.Map<IList<EventViewModel>>(events));
+      }
+
+      return StatusCode((int)HttpStatusCode.NoContent, $"No event found for employee ID: { _employee.EmployeeId }");
```
why are we using  _employee.EmployeeId here instead of the passed in employee id?

nmcmahon-unosquare commented on AdminCore.WebApp/Controllers/EventController.cs:
```diff
@@ -0,0 +1,276 @@
+    [HttpPost]
+    public IActionResult CreateEvent(CreateEventViewModel createEventViewModel)
+    {
+      if (createEventViewModel.EventTypeId == 0)
```
use an enum here for readability rather than hard-coded 0

nmcmahon-unosquare commented on AdminCore.WebApp/Controllers/EventController.cs:
```diff
@@ -0,0 +1,276 @@
+    }
+
+    [HttpPost]
+    public IActionResult CreateEvent(CreateEventViewModel createEventViewModel)
```
This method is a big boi. Break it up into smaller chunks, also consider adding the public holiday check and any other validation checks into a validator. I added this yesterday, an example of it is in the create method in the contract controller

nmcmahon-unosquare commented on AdminCore.WebApp/Controllers/EventController.cs:
```diff
@@ -0,0 +1,276 @@
+      var eventDatesToUpdate = _mapper.Map<EventDateDto>(updateEventViewModel);
+      try
+      {
+        _eventService.IsEventValid(eventDatesToUpdate, updateEventViewModel.IsHalfDay, _employee.EmployeeId);
```
can this validation be moved to the controller?

Jamie-Unosquare replied:
Do you mean the service?

nmcmahon-unosquare replied:
No, the IsEventValid method performs some validation at a service level. Does it need to be in the service or can we move it to the controller?

nmcmahon-unosquare commented on AdminCore.WebApp/Controllers/EventController.cs:
```diff
@@ -0,0 +1,276 @@
+      try
+      {
+        var eventToApprove = _eventService.GetEvent(approveEventViewModel.EventId);
+        if (eventToApprove.EmployeeId != _employee.EmployeeId)
```
factor this admin check out into it's own method and make it throw an exception if an admin tries to approve their own holiday. This will be caught and returned by the catch below.

Jamie-Unosquare committed:
Merge remote-tracking branch 'origin/develop' into Task/Refactor-Controllers-into-All-Purpose-Controller

Jamie-Unosquare committed:
Merge remote-tracking branch 'origin/develop' into Task/Refactor-Controllers-into-All-Purpose-Controller

Jamie-Unosquare committed:
Changes as per PR

Using EmployeeId
Enums for readability
Create Method refactored
Admin Check method
IsEventValid remains in the service layer because it is used by both HolidayController AND EventController until front end transitions fully over to EventController then it can be moved to the controller level
